v2/component/runtime_info.go

```diff
@@ -71,6 +72,27 @@ type runtimeInfo struct {
 	OutputArtifacts  map[string]*outputArtifact
 }
 
+var paramDelimiterRE = regexp.MustCompile(`"BEGIN-KFP-PARAM[[].+?[]]END-KFP-PARAM"`)
+
+func escapeParameters(unescaped string) (string, error) {
+	var jsonEncodeErr error
+	escapeFunc := func(value string) string {
+		value = strings.TrimPrefix(value, `"BEGIN-KFP-PARAM[`)
+		value = strings.TrimSuffix(value, `]END-KFP-PARAM"`)
+		var b []byte
+		b, jsonEncodeErr = json.Marshal(value)
```

Comment:
IIUC escapeFunc is called multiple times, so if at a call, we find that jsonEncodeErr is already non-nil, we should early exit and avoid overriding it with nil again

Reply:
Good point. I've changed it to keep around the last non-nil error instead.